### ISSUE TITLE
feature request: add support to .bin .chd .cso .dax compressed files

### ISSUE THREAD

**Issue by jangix** [aldostools/IRISMAN#41]
Would it be possible to add the support to decompression of the rom file with .cso .dax (for PSP) and chd (for PS1, PS2 and MEME) extensions ?
It would be very useful. Thanks

Compressors Source:
https://github.com/mamedev/mame/blob/master/src/tools/chdman.cpp
https://github.com/unknownbrackets/maxcso

**Comment by aldostools** [comment]
That is a feature that would require to be added to Cobra/Mamba/HEN payload.
As the payload has a limited memory footprint, the feature would have to be implemented in a proxy plugin like rawseciso or netiso.

This is not a simple project (even having the source code of the format available).
I doubt that there are developers willing to expend time and effort on this project.

As a simple workaround, IRISMAN allows users to list zipped PSP ISOs and extract the ISO automatically before mounting it.

**Comment by jangix** [comment]
I imagine the difficulty ... but, for example, having more roms in the .chd format to make them compatible with Irisman and keeping the data compression you should do these conversions for each of them: .chd -> .bin/.cue -> .iso -> .zip/.7z/.rar
Implementing at least .bin/.cue file support (as Multiman already does) would reduce the problem to 3 steps.

**Comment by aldostools** [comment]
I agree with you, but the problem here is get people interested in dedicate time to a complex project like that for free.

If you use ps3netsrv to play PS3/PSP/PSX remotely you can take advantage of the NTFS compression.
Example: Angry Birds.ISO is 161 MB but only uses 20 MB of storage (12%).

According to the info file of MAME 2000 in RetroArch CE, that emulator supports zip, 7z, chd.
MAME 2003 only supports .zip

**Comment by jangix** [comment]
I hope that someday this is possible...
Anyway I didn't know "ps3netsrv", your suggestion helps me a lot thanks!
